src/app/apps/vacunacion/components/lote/lote.component.ts

```diff
@@ -0,0 +1,100 @@
+import { Component, OnInit } from '@angular/core';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { VacunasService } from 'src/app/services/vacunas.service';
+import { Plex } from '@andes/plex';
+
+@Component({
+  selector: 'app-lote',
+  templateUrl: './lote.component.html',
+})
+export class LoteComponent implements OnInit {
+  public vacunaSelected: any;
+  public showSidebar = false;
+  public vacunas$: Observable<any[]>;
+  public lotes$: Observable<any[]>;
+  public lote;
+  public lote2;
+  public vacunasEncontradas;
+  public creando = false;
+  public columns = [
+    {
+      key: 'nombre',
+      label: 'Nombre',
+      sorteable: true,
+      opcional: false
+    }
+  ];
+  public loteSelected =
+    {
+      codigo: '',
+      descripcion: '',
+      habilitado: true,
+      vacuna: null
+    };
+
+  constructor(
+    private vacunasService: VacunasService,
+    private plex: Plex
+  ) { }
+
+  ngOnInit(): void {
+    this.vacunas$ = this.vacunasService.getNomivacVacunas({ habilitado: true, calendarioNacional: 'NO', sort: 'nombre' }).pipe(
+      map(vacunas => {
+        return vacunas.map((v) => {
+          return { _id: v._id, codigo: v.codigo, nombre: v.nombre, snomed_conceptId: v.snomed_conceptId };
+        });
+      })
+    );
+  }
+
+  showInSideBar(vacuna) {
+    if (vacuna) {
+      this.loteSelected.vacuna = vacuna;
+      this.vacunaSelected = vacuna;
+      this.showSidebar = true;
+      this.loadLote();
+    }
+  }
+
+  loadLote() {
+    this.lote = null;
+    this.lotes$ = this.vacunasService.getNomivacLotes({ habilitado: true, sort: 'codigo' }).pipe(
```

Comment:
La API de lotes acepta un parámetro para que solo devuelva los lotes referidos a la vacuna seleccionada.
```suggestion
    this.lotes$ = this.vacunasService.getNomivacLotes({ 
        habilitado: true, 
        vacuna: this.vacunaSelected._id ,
        sort: 'codigo'
    });
```